### PR TITLE
fuzz: fixes oss-fuzz: 9204

### PR DESCRIPTION
**Pull request by anirudhmurali**
Title: Fixes oss-fuzz: 9204

Description: oss-fuzz issue (9204): https://oss-fuzz.com/v2/testcase-detail/5366294281977856
I suppose Envoy doesn't support multiple health checks still. Instead of assertion, have replaced it with throw. This doesn't crash the build. Let me know if any changes.

Risk Level: Low

Testing: Tested unit tests (bazel test //server:server_fuzz_test), built and ran fuzzers with oss-fuzz.

Signed-off-by: Anirudh M m.anirudh18@gmail.com

### CODE REVIEW

File: source/common/upstream/upstream_impl.cc [hunk 1]
```diff
@@ -403,9 +403,13 @@ ClusterSharedPtr ClusterImplBase::create(const envoy::api::v2::Cluster& cluster,
 
   if (!cluster.health_checks().empty()) {
     // TODO(htuch): Need to support multiple health checks in v2.
-    ASSERT(cluster.health_checks().size() == 1);
-    new_cluster->setHealthChecker(HealthCheckerFactory::create(
-        cluster.health_checks()[0], *new_cluster, runtime, random, dispatcher));
+    if (cluster.health_checks().size() != 1)
```
Comment (zuercher) [inline source/common/upstream/upstream_impl.cc]: I believe our style is to use curly braces around both the if and else statements in this case.

File: source/common/upstream/upstream_impl.cc [hunk 2]
```diff
@@ -403,9 +403,14 @@ ClusterSharedPtr ClusterImplBase::create(const envoy::api::v2::Cluster& cluster,
...
-    ASSERT(cluster.health_checks().size() == 1);
-    new_cluster->setHealthChecker(HealthCheckerFactory::create(
-        cluster.health_checks()[0], *new_cluster, runtime, random, dispatcher));
+    if (cluster.health_checks().size() != 1) {
+      throw EnvoyException(
+          fmt::format("Multiple health checks not supported. Health checks count: {}",
+                      cluster.health_checks().size()));
+    } else {
+      new_cluster->setHealthChecker(HealthCheckerFactory::create(
+          cluster.health_checks()[0], *new_cluster, runtime, random, dispatcher, log_manager));
```
Comment (zuercher) [inline source/common/upstream/upstream_impl.cc]: I think you need to delete log_manager.

File: source/common/upstream/upstream_impl.cc [hunk 3]
```diff
@@ -403,9 +403,14 @@ ClusterSharedPtr ClusterImplBase::create(const envoy::api::v2::Cluster& cluster,
 
   if (!cluster.health_checks().empty()) {
     // TODO(htuch): Need to support multiple health checks in v2.
-    ASSERT(cluster.health_checks().size() == 1);
-    new_cluster->setHealthChecker(HealthCheckerFactory::create(
-        cluster.health_checks()[0], *new_cluster, runtime, random, dispatcher));
+    if (cluster.health_checks().size() != 1) {
+      throw EnvoyException(
+          fmt::format("Multiple health checks not supported. Health checks count: {}",
```
Comment (htuch) [inline source/common/upstream/upstream_impl.cc]: Ideally add a unit test in addition to the corpus, thanks.

**Comment by htuch** [comment]
@anirudhmurali can you merge master to workaround the merge conflict here and push? Ta.

**Pull request merged**